sdk/python/kfp/components/structures/components.json_schema.json

```diff
@@ -0,0 +1,402 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "http://kubeflow.org/pipelines/components.json_schema.json",
+
+  "$ref": "#/definitions/PipelineRunSpec",
+
+  "definitions": {
+    "PrimitiveTypes": {
+      "oneOf": [
+        {"type": "string"},
+        {"type": "integer"},
+        {"type": "number"},
+        {"type": "boolean"}
+      ]
+    },
+
+    "TypeType": {
+      "oneOf": [
+        {"type": "string"},
+        {"type": "array", "items": {"$ref": "#/definitions/TypeType"}},
+        {"type": "object", "additionalProperties": {"$ref": "#/definitions/TypeType"}}
+      ]
+    },
+
+    "InputSpec": {
+      "description": "Describes the component input specification",
+      "type": "object",
+      "properties": {
+        "name": {"type": "string"},
+        "type": {"$ref": "#/definitions/TypeType"},
```

Comment:
In the gcp component, I want to provide additional information about the schema of a json object input. Is there anyway that we can provide a field to specify schema information. I'd suggest to provide an annotations list to facilitate to provide additional data to inputs.

Reply:
>I want to provide additional information about the schema of a json object input

It's possible. Please note that it won't reach the executing program. So this metadata will mostly be useful for UI/client validation.

With the current schema you can provide JsonSchema type as follows:
```

inputs:
- name: Input 1
  type:
    JsonSchema: {
      type: object,
      properties:{
        prop1: {type: string},
        prop2: {type: integer},
      }
    }
```

BTW, It's always helpful to get a concrete example of what you want to accomplish. Even more helpful would be a suggestion on how the feature you want will look like in the component definition.